updated packages

khxia opened the pull request:
# Changes

So I updated A LOT of packages. I originally tried to do them one by one starting with Gatsby, but updating Gatsby required me to update a lot of other packages as well. 

I guess this is the full list:
```txt
@material-ui/core               4.11.0  4.12.3  4.12.3 dependencies    https://material-ui.com/

@material-ui/icons              4.9.1   4.11.2  4.11.2 dependencies    https://github.com/mui-org/material-ui/tree/master/packages/material-ui-icons
@svgr/webpack                   5.3.0   5.5.0   5.5.0  dependencies    https://react-svgr.com

classnames                      2.2.6   2.3.1   2.3.1  dependencies    https://github.com/JedWatson/classnames#readme

dayjs                           1.10.4  1.10.7  1.10.7 dependencies    https://day.js.org

eslint                          6.8.0   6.8.0   8.1.0  devDependencies https://eslint.org

eslint-config-standard          14.1.1  14.1.1  16.0.3 devDependencies https://github.com/standard/eslint-config-standard

eslint-plugin-import            2.20.2  2.25.2  2.25.2 devDependencies https://github.com/import-js/eslint-plugin-import

eslint-plugin-promise           4.2.1   4.3.1   5.1.1  devDependencies https://github.com/xjamundx/eslint-plugin-promise

eslint-plugin-react             7.19.0  7.26.1  7.26.1 devDependencies https://github.com/yannickcr/eslint-plugin-react

eslint-plugin-react-hooks       3.0.0   3.0.0   4.2.0  devDependencies https://reactjs.org/

eslint-plugin-standard          4.0.1   4.1.0   5.0.0  devDependencies https://github.com/standard/eslint-plugin-standard#readme
gatsby-image                    2.3.1   2.11.0  3.11.0 dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-image#readme
gatsby-plugin-manifest          2.3.3   2.12.1  4.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-manifest#readme
gatsby-plugin-material-ui       2.1.10  2.1.10  4.0.3  dependencies    https://github.com/hupe1980/gatsby-plugin-material-ui#readme
gatsby-plugin-react-helmet      3.2.1   3.10.0  5.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-react-helmet#readme
gatsby-plugin-sharp             2.5.3   2.14.4  4.0.1  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-sharp#readme
gatsby-remark-autolink-headers  2.2.1   2.11.0  5.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-autolink-headers#readme
gatsby-remark-copy-linked-files 2.2.1   2.10.0  5.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-copy-linked-files#readme
gatsby-remark-images            3.2.1   3.11.1  6.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-images#readme
gatsby-remark-prismjs           3.4.1   3.13.0  6.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-prismjs#readme
gatsby-source-filesystem        2.2.2   2.11.1  4.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-filesystem#readme
gatsby-transformer-remark       2.7.1   2.16.1  5.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-remark#readme
gatsby-transformer-sharp        2.4.3   2.12.1  4.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-sharp#readme
gatsby-transformer-yaml         2.4.10  2.11.0  4.0.0  dependencies    https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-yaml#readme
gh-pages                        2.2.0   2.2.0   3.2.3  devDependencies https://github.com/tschaub/gh-pages

husky                           4.2.3   4.3.8   7.0.4  devDependencies https://typicode.github.io/husky

prettier                        2.0.2   2.4.1   2.4.1  devDependencies https://prettier.io

react                           16.13.1 16.14.0 17.0.2 dependencies    https://reactjs.org/

react-dom                       16.13.1 16.14.0 17.0.2 dependencies    https://reactjs.org/

react-helmet                    5.2.1   5.2.1   6.1.0  dependencies    https://github.com/nfl/react-helmet#readme

react-slick                     0.27.10 0.27.14 0.28.1 dependencies    https://react-slick.neostack.com

sharp                           0.25.2  0.25.4  0.29.2 dependencies    https://github.com/lovell/sharp

svgo                            1.3.2   1.3.2   2.7.0  dependencies    https://github.com/svg/svgo

svgo-loader                     2.2.1   2.2.2   3.0.0  dependencies    https://github.com/pozadi/svgo-loader

```
The upgrades of some of these packages have also required me to manually install some new packages:
- @emotion/react
- path-browserify -> this also had to be added to webpack as a fallback due to polyfill no longer being supported??

In particular, `husky`, `webpack`, and `eslint` updates gave me a lot of headaches - and I'm not sure if I have resolved all of the issues. 

Currently, I am experiencing some really bad latency issues whenever I click a button to change pages, and I wanted to make sure if this will be an issue when the website is built. So more investigation needs to be done on this front too.

This was my first time dealing with a lot of these libraries, so any help is appreciated. (Help me senpai)

Oh, and I also fixed some bugs that I did not manage to catch in #262
## Type of change

Please delete options that are not relevant.

- [ ] Content Update (non-breaking change which updates the content of the website)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Related Issues

#203

khxia commented:
Ok yeah something is very wrong with this. Will need to investigate.

khxia commented:
Ok, although this current version doesn't seem break anything in the website, the build logs (from netlify deploy-preview) still has some warnings that I am not sure if we should address.
```txt
 warning " > eslint-plugin-react@7.26.1" has incorrect peer dependency "eslint@^3 \|\| ^4 \|\| ^5 \|\| ^6 \|\| ^7".
--
  | warning " > eslint-plugin-react-hooks@4.2.0" has incorrect peer dependency "eslint@^3.0.0 \|\| ^4.0.0 \|\| ^5.0.0 \|\| ^6.0.0 \|\| ^7.0.0".
  | warning "gatsby > @typescript-eslint/eslint-plugin > tsutils@3.21.0" has unmet peer dependency "typescript@>=2.8.0 \|\| >= 3.2.0-dev \|\| >= 3.3.0-dev \|\| >= 3.4.0-dev \|\| >= 3.5.0-dev \|\| >= 3.6.0-dev \|\| >= 3.6.0-beta \|\| >= 3.7.0-dev \|\| >= 3.7.0-beta".
  | warning "gatsby > eslint-plugin-graphql > graphql-config > @endemolshinegroup/cosmiconfig-typescript-loader > ts-node@9.1.1" has unmet peer dependency "typescript@>=2.7".
  | warning " > gatsby-plugin-material-ui@3.0.1" has unmet peer dependency "@material-ui/styles@>=4.0.0".
  | warning " > gatsby-plugin-material-ui@3.0.1" has incorrect peer dependency "gatsby@^3.0.0".
  | warning "gatsby-plugin-material-ui > babel-preset-gatsby@1.14.0" has unmet peer dependency "@babel/core@^7.11.6".
  | warning "gatsby-plugin-material-ui > babel-preset-gatsby@1.14.0" has unmet peer dependency "core-js@^3.0.0".
  | warning " > slick-carousel@1.8.1" has unmet peer dependency "jquery@>=1.8.0".
  | warning " > eslint-config-standard@16.0.3" has incorrect peer dependency "eslint@^7.12.1".
  | warning " > eslint-plugin-promise@5.1.1" has incorrect peer dependency "eslint@^7.0.0".
```

jodymlin commented:
![image](https://user-images.githubusercontent.com/44387428/143384691-bc9edea2-01b1-43a5-a8ee-92c87cdc1067.png)
I get this error when I try to build it locally after running `yarn install`. 

We should also probably upgrade the material-ui dependencies since the it seems our versions conflict. 

Also take a look at this [blog post](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v3-to-v4/) about upgrading gatsby from v3 to v4. Looks like a lot of breaking changes were made.

TimothyGu commented on package.json:
```diff
@@ -10,50 +10,53 @@
+    "@svgr/webpack": "^5.5.0",
+    "classnames": "^2.3.1",
+    "dayjs": "^1.10.7",
+    "false": "^0.0.4",
```
What's this for?

khxia replied:
I honestly have no idea, not sure why this package was added, seems to be a mistake on my part

khxia commented:
After running a codemod for the new gatsby-image api, got these messages. Will be addressing them. Am also going to paste the messages here to document it somewhere.

```txt
It appears you're passing a variable to your image component. We haven't changed it, but we have updated it to use the new GatsbyImage component. Please check src\components\PhotoPage\Cropper.js manually.
It appears you're passing a variable to your image component. We haven't changed it, but we have updated it to use the new GatsbyImage component. Please check src\components\PhotoPage\Cropper.js manually.
It appears you're passing src or srcSet directly. This API is no longer exposed, please update src\components\SvgImg.js manually.
It appears you are referencing the image component in some way. We've updated the reference, but you will want to verify src\components\EventCard\EventCard.js manually.
maxWidth is no longer supported for fluid (now fullWidth) images. We've updated the query in src\components\EventCard\EventCard.js to use a constrained image instead.
maxHeight is no longer supported for fluid (now fullWidth) images. It's been removed from your query in src\components\EventCard\EventCard.js.
It appears you're passing src or srcSet directly. This API is no longer exposed, please update src\components\HomePage\Carousel.js manually.
```

khxia commented:
The SvgImg component is now no longer being used. (it can no longer be used because of API changes). The new way I tried to render svg's is using the `StaticImage` component, which doesn't render the svg's at the best quality. The homepage banner got  a decrease in image quality while all the other images got an increase in image quality.
